vm/src/stdlib/ctypes/base.rs

```diff
@@ -150,12 +165,45 @@ impl Debug for PyCSimple {
 }
 
 #[pyclass(flags(BASETYPE))]
-impl PyCSimple {}
+impl PyCSimple {
+    #[pymethod(magic)]
+    pub fn __init__(&self, value: OptionalArg, vm: &VirtualMachine) -> PyResult<()> {
+        if let Some(ref v) = value.into_option() {
+            let content = set_primitive(self._type_.as_str(), v, vm)?;
+            self.value.store(content);
+        } else {
+            self.value.store(match self._type_.as_str() {
+                "c" | "u" => PyObjectRef::from(vm.ctx.new_bytes(vec![0])),
+                "b" | "B" | "h" | "H" | "i" | "I" | "l" | "q" | "L" | "Q" => PyObjectRef::from(vm.ctx.new_int(0)),
+                "f" | "d" | "g" => PyObjectRef::from(vm.ctx.new_float(0.0)),
+                "?" => PyObjectRef::from(vm.ctx.new_bool(false)),
+                _ => vm.ctx.none(), // "z" | "Z" | "P"
+            });
+        }
+        Ok(())
+    }
 
-pub fn extend_module_nodes(vm: &VirtualMachine, module: &Py<PyModule>) {
-    let ctx = &vm.ctx;
-    extend_module!(vm, module, {
-        "_CData" => PyCData::make_class(ctx),
-        "_SimpleCData" => PyCSimple::make_class(ctx),
-    })
+    #[pygetset(name = "value")]
+    pub fn value(instance: PyObjectRef, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+        let cls = instance.class();
+        let subcls_vec = cls.subclasses.read();
+        for subcls in subcls_vec.iter() {
+            println!("subcls {}", subcls);
+        }
+        println!("value {}", cls.name().to_string());
+        let zelf: &Py<Self> = instance.downcast_ref().ok_or_else(|| {
```

Comment:
Currently it always fails because the payload is `PyBaseObject` instead of `PyCSimple`. Adding a `Constructor` for this type will fix the problem. Don't forget to return a correct PyCSimple object in Constructor.